src/parsers/perf/perfparser.cpp

```diff
@@ -838,6 +841,24 @@ class PerfParserPrivate : public QObject
             }
         }
 
+        const auto costAggregation = Settings::instance()->costAggregation();
+
+        if (costAggregation == Settings::CostAggregation::ByThread) {
```

Comment:
the code is duplicated below, merge this and say something like `costAggregation == ByThread || costAggregatoin == ByProcess`

Reply:
also, this whole process of giving the top-level entries pretty names here is bogus - that means the topDownResults would have the non-prettied names as leaf nodes, which is undesired. they should also be root nodes, and also get prettified

similarly, we need to prettify or skip these entries when we buildCallerCalleeResult etc. above.

to fix this, apply the pretty names early (see suggestions below). but one way or another the handling for the buildTopDownResult needs to be rethought.